CONTRIBUTING.md

```diff
@@ -78,6 +79,66 @@ use only the standard libraries, or the third-party packages we have decided to
 
 For rationale, check out the post [The Case Against Third Party Libraries](http://blog.gopheracademy.com/advent-2014/case-against-3pl/).
 
+### Writing unit tests
+Unit tests for Go code should be in the same package as the code itself.
+It should be in a file named `filename_test.go` with the package name `mypackage_test` as the name of the package.
+The tests should test functionality through public functions and methods.
+
+If access to internal members or functions is needed, then the test file should be named `filename_internal_test.go` and use the same package name as the package itself.
+The tests written in this file will have access to the internals of the package.
+This file can also be used to create testing-specific functions that can be exposed to the tests in `mypackage_test`.
+
+Pull requests require tests to be written for them and must pass our code coverage checks.
+
+### Public functions and methods
+When writing code, try to avoid making functions and methods used by the code public.
+When creating a library that needs to be used by multiple packages within the Flux repository, an internal package is preferred.
+A public function or method should only be used if using the Flux library would require the function or method to be public.
+
+### Comments and documentation
+Every new public function or method requires a valid GoDoc.
+Comments within the code should be written with proper capitalization and punctuation at the end of the sentence.
+
```

Comment:
Can we link to Go's documentation on how to write proper comments? I think this might be a good one: https://blog.golang.org/godoc-documenting-go-code

Reply:
Yes and I should find a place to link effective go somewhere.